Problem: link failure building czmq_selftest and zmakecert

Building either `czmq_selftest` or `zmakecert` results in this linker error on my Ubuntu build (static/prefix).
```
  CC       src/src_zmakecert-zmakecert.o
  CC       src/src_czmq_selftest-czmq_selftest.o
  CCLD     src/libczmq.la
  CCLD     src/czmq_selftest
  CCLD     src/zmakecert

/usr/bin/ld: /home/eric/bs-prefix//lib/libzmq.a(usrsrc_libzmq_la-/bindecoder_allocators.o)/ld:: undefined / referencehome/eric to/bs- symbolprefix/ 'ceil@@lib/libzmq.aGLIBC_2.2.5(src_libzmq_la'
/-decoder_allocators.o/lib/): undefinedx86_64- reference linux-gnuto /libm.so.6symbol 'ceil@@GLIBC_2.2.5'
//lib/x86_64: -linuxerror adding-gnu symbols/libm.so.6::  errorDSO missing adding from  command symbols:line
DSO missing from command line

collect2: error: ld returned 1 exit status
collect2: error: ld returned 1 exit status
make[1]: *** [src/zmakecert] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [src/czmq_selftest] Error 1
```
It appears that `libm` is either missing or improperly ordered in the dependencies for these two builds.

I have almost the same issue, but with pthread on Debian 8. When I run `./configure --disable-shared` I get the error when running `make`: 

    CCLD     src/makecert
    CCLD     src/test_zgossip
    /usr/bin/ld: src/.libs/libczmq.a(src_libczmq_la-zactor.o): undefined reference to symbol      'pthread_create@@GLIBC_2.2.5'
    //lib/x86_64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
    collect2: error: ld returned 1 exit status
    Makefile:1062: recipe for target 'src/makecert' failed
    make[1]: *** [src/makecert] Error 1
    make[1]: *** Waiting for unfinished jobs....
    /usr/bin/ld: src/.libs/libczmq.a(src_libczmq_la-zactor.o): undefined reference to symbol       'pthread_create@@GLIBC_2.2.5'
    //lib/x86_64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
    collect2: error: ld returned 1 exit status
    Makefile:1068: recipe for target 'src/test_zgossip' failed
    make[1]: *** [src/test_zgossip] Error 1

A workaround I found is `make LDFLAGS="-lpthread"`.

--- a/src/Makemodule.am
+++ b/src/Makemodule.am
@@ -6,7 +6,7 @@ program_libs = src/libczmq.la ${project_libs}
 
 # Programs need to link the c++ runtime if everything was compiled statically.
 if !ENABLE_SHARED
-program_libs += -lstdc++
+program_libs += -lstdc++ -lm
 endif
 
 lib_LTLIBRARIES += src/libczmq.la

